### ISSUE TITLE
ValueError: target size and input size have a mismatch

### ISSUE THREAD

**Issue by yhnnstshy** [milesial/Pytorch-UNet#164]
Please find the runtime error below: 

ValueError: Target size (torch.Size([1, 3, 256, 256])) must be the same as input size (torch.Size([1, 1, 256, 256]))

This occurs at line 79 in the train.py file (loss = criterion(masks_pred, true_masks)). It seems like the loss doesn't accept input and output channel mismatch. Am I the only one having this issue?

**Comment by milesial** [comment]
You are loading your target masks wrong, please search in the closed issues there are similar ones.

**Comment by yhnnstshy** [comment]
Thank you for responding. I tried the suggestions in the closed issues. Is there anything in particular that I should be doing when I'm loading the target images? My target images are RGB images with three channels with 255,255,255 when the label is positive and 0,0,0 for background (I made them using ITK), and I used the default dataset.py code to load them. The output from dataset.py has a mask that is transposed to CWH where C=3. 

Any help would be appreciated. Thanks.
